types/src/get_with_proof.rs

```diff
@@ -396,17 +406,21 @@ fn verify_get_events_by_access_path_resp(
         expected_seq_nums.len(),
         events_with_proof.len(),
     );
-    itertools::zip_eq(events_with_proof, expected_seq_nums)
-        .map(|(e, seq_num)| {
-            e.verify(
-                ledger_info,
-                expected_event_key,
-                seq_num,
-                e.transaction_version,
-                e.event_index,
-            )
-        })
-        .collect::<Result<Vec<_>>>()?;
+    if let Some(expected_event_key) = expected_event_key_opt {
+        itertools::zip_eq(events_with_proof, expected_seq_nums)
+            .map(|(e, seq_num)| {
+                e.verify(
+                    ledger_info,
+                    &expected_event_key,
+                    seq_num,
+                    e.transaction_version,
+                    e.event_index,
+                )
+            })
+            .collect::<Result<Vec<_>>>()?;
+    } else if !events_with_proof.is_empty() {
+        bail!("Bad events_with_proof: nonempty event list for nonexistent account")
```

Comment:
@wqfish: Does this error case make sense? This is definitely different than the behavior before this PR, but it seems like proof of a non-empty event sequence on a nonexistent account should always fail.

Reply:
Ugh this is one of the most complex function we've written...

I think if `expected_event_key_opt` is `None`, then `seq_num_upper_bound` is 0, then `expected_seq_nums` at line 389 will be an empty vector (the first branch `cursor >= seq_num_upper_bound`), then the `ensure` at line 403 will error, so the old code would fail as well. Took me a while to read the code...